modules/fanAdapter.js

```diff
@@ -0,0 +1,383 @@
+      gl.gs(gl.VERTEX_SHADER, gl.LOW_FLOAT),
+    ];
+    var frag = [
+      gl.gs(gl.FRAGMENT_SHADER, gl.HIGH_FLOAT),
+      gl.gs(gl.FRAGMENT_SHADER, gl.MEDIUM_FLOAT),
+      gl.gs(gl.FRAGMENT_SHADER, gl.LOW_FLOAT),
+    ];
+
+    window.evg = debugInfo
+      ? gl.gp(debugInfo.UNMASKED_RENDERER_WEBGL)
+      : gl.gp(gl.RENDERER);
+
+    hash = [
+      vertex,
+      frag,
+      gl.getContextAttributes().antialias,
+      gl.gp(gl.VERSION),
+      gl.gp(gl.SHADING_LANGUAGE_VERSION),
+      gl.gp(gl.VENDOR),
+      gl.gp(gl.ALIASED_LINE_WIDTH_RANGE),
+      gl.gp(gl.RED_BITS),
+      gl.gp(gl.GREEN_BITS),
+      gl.gp(gl.BLUE_BITS),
+      gl.gp(gl.ALPHA_BITS),
+      gl.gp(gl.DEPTH_BITS),
+      gl.gp(gl.STENCIL_BITS),
+      gl.gp(gl.MAX_RENDERBUFFER_SIZE),
+      gl.gp(gl.MAX_COMBINED_TEXTURE_IMAGE_UNITS),
+      gl.gp(gl.MAX_CUBE_MAP_TEXTURE_SIZE),
+      gl.gp(gl.MAX_FRAGMENT_UNIFORM_VECTORS),
+      gl.gp(gl.MAX_TEXTURE_IMAGE_UNITS),
+      gl.gp(gl.MAX_TEXTURE_SIZE),
+      gl.gp(gl.MAX_VARYING_VECTORS),
+      gl.gp(gl.MAX_VERTEX_ATTRIBS),
+      gl.gp(gl.MAX_VERTEX_TEXTURE_IMAGE_UNITS),
+      gl.gp(gl.MAX_VERTEX_UNIFORM_VECTORS),
+      gl.gp(gl.ALIASED_LINE_WIDTH_RANGE),
+      gl.gp(gl.ALIASED_POINT_SIZE_RANGE),
+      gl.gp(gl.MAX_VIEWPORT_DIMS),
+      gl.gs(gl.FRAGMENT_SHADER, gl.HIGH_INT),
+      gl.getSupportedExtensions(),
+      debugInfo ? gl.gp(debugInfo.UNMASKED_RENDERER_WEBGL) : gl.gp(gl.RENDERER),
+      drawBuffers ? gl.gp(drawBuffers.MAX_DRAW_BUFFERS_WEBGL) : 1,
+      anisotropy ? gl.gp(anisotropy.MAX_TEXTURE_MAX_ANISOTROPY_EXT) : 0,
+    ];
+  }
+
+  return MD5(JSON.stringify(hash)).toString();
+}
+
+/**
+ * Generate stable user id
+ *
+ * @returns {String} userId
+ */
+function generateUserId() {
+  var flavors = [
+    // Blink and some browsers on iOS
+    'chrome',
+
+    // Safari on macOS
+    'safari',
+
+    // Chrome on iOS (checked in 85 on 13 and 87 on 14)
+    '__crWeb',
+    '__gCrWeb',
+
+    // Yandex Browser on iOS, macOS and Android (checked in 21.2 on iOS 14, macOS and Android)
+    'yandex',
+
+    // Yandex Browser on iOS (checked in 21.2 on 14)
+    '__yb',
+    '__ybro',
+
+    // Firefox on iOS (checked in 32 on 14)
+    '__firefox__',
+
+    // Edge on iOS (checked in 46 on 14)
+    '__edgeTrackingPreventionStatistics',
+    'webkit',
+
+    // Opera Touch on iOS (checked in 2.6 on 14)
+    'oprt',
+
+    // Samsung Internet on Android (checked in 11.1)
+    'samsungAr',
+
+    // UC Browser on Android (checked in 12.10 and 13.0)
+    'ucweb',
+    'UCShellJava',
+
+    // Puffin on Android (checked in 9.0)
+    'puffinDevice',
+  ];
+
+  var flavor = null;
+  for (var i = 0; i < flavors.length; i++) {
+    if (window[flavors[i]] && typeof window[flavors[i]] === 'object') {
+      flavor = flavors[i];
+    }
+  }
+
+  var screen = [
+    parseInt(window.screen.width),
+    parseInt(window.screen.height),
+    window.screen.availHeight,
+    window.screen.availWidth,
+  ];
+
+  var hash = [
```

Comment:
You might find some more ideas for your fingerprint at https://amiunique.org/fingerprint or
https://github.com/duckduckgo/tracker-radar/blob/main/build-data/generated/api_fingerprint_weights.json :) We're not able to accept this pr

Reply:
I have removed the problematic code and applied all recommendations.